code/game/mecha/mech_fabricator.dm

```diff
@@ -208,67 +267,47 @@
 			return process_queue()
 		else
 			return
-	temp = null
 	while(D)
-		if(stat&(NOPOWER|BROKEN))
+		if(stat & (NOPOWER|BROKEN))
 			return FALSE
-		if(!check_resources(D))
+		if(!can_build(D))
 			atom_say("Not enough resources. Queue processing stopped.")
-			temp = {"<span class='alert'>Not enough resources to build next part.</span><br>
-						<a href='?src=[UID()];process_queue=1'>Try again</a> | <a href='?src=[UID()];clear_temp=1'>Return</a><a>"}
 			return FALSE
 		remove_from_queue(1)
 		build_part(D)
 		D = listgetindex(queue, 1)
 	atom_say("Queue processing finished successfully.")
 
-/obj/machinery/mecha_part_fabricator/proc/list_queue()
-	var/output = "<b>Queue contains:</b>"
-	if(!istype(queue) || !queue.len)
-		output += "<br>Nothing"
-	else
-		output += "<ol>"
-		var/i = 0
-		for(var/datum/design/D in queue)
-			i++
-			var/obj/part = D.build_path
-			output += "<li[!check_resources(D)?" style='color: #f00;'":null]>"
-			output += initial(part.name) + " - "
-			output += "[i>1?"<a href='?src=[UID()];queue_move=-1;index=[i]' class='arrow'>&uarr;</a>":null] "
-			output += "[i<queue.len?"<a href='?src=[UID()];queue_move=+1;index=[i]' class='arrow'>&darr;</a>":null] "
-			output += "<a href='?src=[UID()];remove_from_queue=[i]'>Remove</a></li>"
-
-		output += "</ol>"
-		output += "\[<a href='?src=[UID()];process_queue=1'>Process queue</a> | <a href='?src=[UID()];clear_queue=1'>Clear queue</a>\]"
-	return output
-
+// Syncs with the local R&D console in the area
 /obj/machinery/mecha_part_fabricator/proc/sync()
-	temp = "Updating local R&D database..."
-	updateUsrDialog()
+	add_wait_message("Syncing with local R&D database...", 30)
+	SSnanoui.update_uis(src) // To get the wait message to show up
 	sleep(30) //only sleep if called by user
 	var/area/localarea = get_area(src)
 
 	for(var/obj/machinery/computer/rdconsole/RDC in localarea.contents)
 		if(!RDC.sync)
 			continue
 		RDC.files.push_data(files)
-		temp = "Processed equipment designs.<br>"
-		//check if the tech coefficients have changed
-		temp += "<a href='?src=[UID()];clear_temp=1'>Return</a>"
-
-		updateUsrDialog()
+		SSnanoui.update_uis(src)
 		atom_say("Successfully synchronized with R&D server.")
-		return
+		return TRUE

```

Comment:
I don't think there's value in adding return values (ha!) when nobody is checking them.
It's better to leave them bare, and only populate once you know what semantics you *want* to act on (and document that semantics).

Reply:
fair enough